File: Classe extendida Fila e Pilha/C++/fila.cpp
#include <iostream>
#include <queue>
#include "lista.h"
#include "fila.h"

int Fila::first(){
    if(this->fila.empty()){
        std::cout << "Fila vazia" << std::endl;
        return -1;
    } else return this->fila.front();
}

void Fila::insere(int value){
    this->fila.push(value);
    this->incElem();
}

void Fila::retira(){
    if(this->fila.empty()){
        std::cout << "Fila vazia" << std::endl;
        return;
    } else this->fila.pop(); this->decElem();
}

File: Classe extendida Fila e Pilha/Java/Fila.java
import java.util.ArrayDeque;

public class Fila extends Lista {
    private ArrayDeque<Integer> fila;

    public Fila(){
        super();
        this.fila = new ArrayDeque<>();
    }

    public int first(){
        if (this.fila.isEmpty()){
            System.out.println("fila vazia\n");
            return -1;
        } else return this.fila.getFirst();
    }

    public void insere(int value){
        this.fila.offer(value);
        this.incElem();
    }

    public int retira(){
        this.decElem();
        return this.fila.poll();
    }
}